src/runtime/mini-prompt-api.js

```
@@ -0,0 +1,240 @@
+    );
+    titleContainerDiv.classList.add('swg-mini-prompt-title-container');
+    titleContainerDiv./*OK*/ innerHTML = TITLE_CONTAINER_DIV_HTML.replace(
+      '/$theme$/g',
+      theme
+    ).replace('$textContent$', textContent);
+    const closeContainerDiv = createElement(
+      this.doc_.getWin().document,
+      'div',
+      {}
+    );
+    closeContainerDiv.classList.add('swg-mini-prompt-close-button-container');
+    closeContainerDiv./*OK*/ innerHTML = CLOSE_CONTAINER_DIV_HTML.replace(
+      '$theme$',
+      theme
+    );
+    miniPromptDiv.appendChild(titleContainerDiv);
+    miniPromptDiv.appendChild(closeContainerDiv);
+    this.doc_.getWin().document.body.appendChild(miniPromptDiv);
+
+    // Handle events and logging for the various sub-components.
+    const clickFun = () => {
+      this.logClick_(options.autoPromptType);
+      if (typeof options.callback === 'function') {
+        options.callback();
+      }
+      setStyle(miniPromptDiv, 'visibility', 'hidden');
+    };
+    titleContainerDiv.addEventListener('click', clickFun);
+
+    const closeFun = () => {
+      setStyle(miniPromptDiv, 'visibility', 'hidden');
+      this.logClose_(options.autoPromptType);
+    };
+    closeContainerDiv.addEventListener('click', closeFun);
+
+    this.logImpression_(options.autoPromptType);
+  }
+
+  /**
+   * Logs an impression of the mini prompt.
+   * @param {!AutoPromptType} autoPromptType
+   */
+  logImpression_(autoPromptType) {
+    let event;
+    if (autoPromptType === AutoPromptType.CONTRIBUTION) {
+      event = AnalyticsEvent.IMPRESSION_SWG_CONTRIBUTION_MINI_PROMPT;
+    } else if (autoPromptType === AutoPromptType.SUBSCRIPTION) {
+      event = AnalyticsEvent.IMPRESSION_SWG_SUBSCRIPTION_MINI_PROMPT;
+    }
+
+    if (event) {
+      this.eventManager_.logSwgEvent(event, false);
+    }
+  }
+
+  /**
+   * Logs a click of the mini prompt.
+   * @param {!AutoPromptType} autoPromptType
+   */
+  logClick_(autoPromptType) {
```

Could you add test coverage for this method?

---

There was test coverage, but there's some weird silent error with the mini-prompt-api-test after I modified the html insertion. Looking into why parts of the test aren't executing now.

---

The tests should `await` the result of `expectMiniPromptCreated()`

---

Fixed.